catch-uncommitted.sh

```diff
@@ -18,6 +18,13 @@ while [ $# -gt 0 ]; do
 		--ignore-space-at-eol)
 			ignore_eol="--ignore-space-at-eol"
 		;;
+		--excludes=*)
+			while IFS=',' read -ra paths; do
+				for path in "${paths[@]}"; do
+					extra_args="${extra_args} "":(exclude)${path}"""
```

Comment:
This looks like it might struggle with spaces in filenames, `extra_args` should be treated as an array until the value is needed.

```sh
extra_args=()
# ...
while ...; do
  # ...
  extra_args+=("(exclude)${path}")
  # ...
# ...
git diff HEAD -- . "${extra_args[@]}"
# ...
```

Reply:
My own not-necessarily-helpful 2 cents is that I wish we used less shell scripts and more of... I suppose Javascript, because I think it is the language that balena uses the most, though for scripts like this perhaps Python would be a bit friendlier.

Reply:
@pdcastro shell scripts are great but the whole world should just target zsh; then the whole loop just becomes something like…

```zsh
":(exclude)"${^${(ps:,:)${key#--exclude}}}
```

😁

Reply:
Can we change this shell script to be ran by `/bin/bash` instead of `/bin/sh`? Never had a good time dealing with arrays in `sh`.

Reply:
balena CI runs some repo tests (including `catch-uncommitted`) on Alpine Linux if I am not mistaken. Mostly likely it has `/bin/bash` installed. I guess we will quickly find out if it doesn't. :-)

Reply:
Looks like the alpine tests passed 🎉